pkg/util/dynamichelper/gvrresolver_test.go

```diff
@@ -195,17 +199,35 @@ func TestFindGVR(t *testing.T) {
 		},
 	}
 
-	for _, tt := range tests {
+	for index, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
 			r := &gvrResolver{apiresources: tt.resources}
 
 			got, err := r.Resolve(tt.kind, "")
-			if !reflect.DeepEqual(err, tt.wantErr) {
-				t.Errorf("got: %#v, expected: %#v", err, tt.wantErr)
+			if err != nil {
+				for _, i := range deep.Equal(err.Error(), tt.wantErr.Error()) {
```

Comment:
I'm noticing this was code that existed before, but `deep.Equal` or `reflect.DeepEqual`, etc on type Error is not best practice, linters typically fail on stuff like this: https://github.com/golang/lint/issues/232

Reply:
Thanks for the feedback! What's the recommended method for truly fixing this?

Reply:
That's discussed in the PR I linked :)